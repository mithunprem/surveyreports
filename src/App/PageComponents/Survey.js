import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Col, Card, CardHeader, CardBody, CardText, Button } from 'reactstrap';
import { withRouter } from 'react-router-dom';
import { SURVEY_DETAILS_URL } from '../Constants';

class Survey extends Component {

  openSurveyDetailsPage = () => {
    const { history, survey } = this.props;

    history.push({
      pathname: SURVEY_DETAILS_URL,
      state: { url: survey.url }
    });
  }

  render() {
    const {
      name, submitted_response_count,
      participant_count, response_rate
    } = this.props.survey;

    return (
      <Fragment>
        <Col xs={8} sm="auto" className="p-0 m-3">
          <Card>
            <CardHeader tag="h5">{name}</CardHeader>
            <CardBody>
              <CardText>
                Participant Count : {participant_count}
              </CardText>
              <CardText>
                Submitted Response Count : {submitted_response_count}
              </CardText>
              <CardText>
                Response Rate : {(response_rate*100).toFixed(2)}%
              </CardText>
              <Button size="sm" onClick={this.openSurveyDetailsPage}>
                View Survey Details
              </Button>
            </CardBody>
          </Card>
        </Col>
      </Fragment>
    )
  }
}

Survey.propTypes = {
  history: PropTypes.object,
  survey: PropTypes.object
}

export default withRouter(Survey);
