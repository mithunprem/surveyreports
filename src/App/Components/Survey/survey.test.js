import React from 'react';
import { create } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import Survey from '../../Components/Survey';

describe("Survey", () => {
  const surveyData = {
    name: 'Test survey',
    submitted_response_count: 6,
    participant_count: 6,
    response_rate: 1
  };

  const history = {};

  test("should render without throwing an error", () => {
    const survey = create(
      <BrowserRouter>
        <Survey survey={surveyData} history={history}/>
      </BrowserRouter>).toJSON();
    expect(survey).toMatchSnapshot();
  });
});
