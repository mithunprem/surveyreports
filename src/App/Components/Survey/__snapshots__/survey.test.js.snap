// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Survey should render without throwing an error 1`] = `
<div
  className="p-0 m-3 col-8 col-sm-auto"
>
  <div
    className="card"
  >
    <h5
      className="card-header"
    >
      Test survey
    </h5>
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        Participant Count : 
        6
      </p>
      <p
        className="card-text"
      >
        Submitted Response Count : 
        6
      </p>
      <p
        className="card-text"
      >
        Response Rate : 
        100.00
        %
      </p>
      <button
        aria-label="View Survey Details Button"
        className="btn btn-secondary btn-sm"
        onClick={[Function]}
        type="button"
      >
        View Survey Details
      </button>
    </div>
  </div>
</div>
`;
